test/Streamly/Test/Data/Unfold.hs

```diff
@@ -111,8 +109,10 @@ fromStream =
     property
         $ \list ->
               testUnfoldD
-                  UF.fromStream
-                  (S.fromList list :: SerialT Identity Int)
+                  UF.fromStreamK
+                    (S.toStreamK
+                        (S.unfold UF.fromList list :: S.Stream Identity Int)
```

Comment:
Implement the following using `fromStreamD` in Internal.Data.Unfold module.:

```
fromStream :: Applicative m => Unfold m (Stream m a) a
```

And use that here. We cannot fromStreamK, that is purely internal.